File: py/ex4.py
#!/bin/python

## Exercício 2.7
import pandas as pd
import numpy as np

def main():
  valor = float(input("Qual o valor? "))

  juros(valor)


def juros(valor):
  parcelas = [1, 3, 6, 9, 12]
  total_juros = np.array([1.0, 1.1, 1.15, 1.2, 1.25])*valor

  df = pd.DataFrame({"Valor da Divida":total_juros,
                     "Valor dos Juros":total_juros-valor,
                     "Quantidade de Parcelas":parcelas,
                     "Valor da Parcela":total_juros/parcelas})

  print(df)


if __name__ == "__main__":
  main()

File: R/ex4.R
#!/bin/Rscript

## Exercício 2.7
main <- function()
{
  cat("Qual o valor? ")
  valor <- as.numeric(readLines("stdin", n = 1))
  
  juros(valor)
}

juros <- function(valor)
{
  total_juros <- c(valor,
                   valor*1.1,
                   valor*1.15,
                   valor*1.2,
                   valor*1.25)
  
  parcelas <- c(1,3,6,9,12)
  
  df <- data.frame(Valor_da_Divida = total_juros,
                   Valor_dos_Juros = total_juros - valor,
                   Quantidade_de_Parcelas = parcelas,
                   Valor_da_Parcela = total_juros / parcelas)
  
  return(df)
}

main()
